### PR TITLE
Fixes #3253 Add Image Credit to Photo Galleries.

### LINKED ISSUE #3253: Add Image Credit to Photo Galleries
## Motivation
The Photo Gallery paragraph type does not presently show image photo credit. I'd like Photo Galleries to show image credits for images that have them.

## Proposed Resolution
Add photo credit to the image gallery paragraph.

### Describe the solution you'd like
Add `field_az_credit` to the display of photo galleries.

### Describe alternatives you've considered
Not having the photo credit displayed.

### PR DESCRIPTION
**Pull request by tadean**
## Description
This PR changes the display of photo galleries to render credit if one is available.

## Related issues
Closes #3253 

## How to test

1. Add a Caption and Credit to an image
2. Place those images into a photo gallery paragraph
3. Verify that the credit displays in both the slider and grid view.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tadean** [comment]
**For discussion:**

- Determine if this proposal is something we should implement
- If so, should the credit be displayed always, or only if a caption is also present? Currently this PR only displays the credit in a photo gallery if a caption also exists.

**Comment by joeparsons** [comment]
> **For discussion:**
> 
> * Determine if this proposal is something we should implement
> * If so, should the credit be displayed always, or only if a caption is also present? Currently this PR only displays the credit in a photo gallery regardless of if a caption exists.

I think the way you have this implemented currently (always display credit if it exists regardless of whether caption exists) makes sense.

**Review by joeparsons — DISMISSED** [review]

**Review by trackleft — DISMISSED** [review]

File: modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig [hunk 1]
```diff
@@ -105,10 +105,17 @@
...
 
-                    {# Render the caption, if one exists. #}
-                    {% if item.entity.field_az_caption.value %}
+                    {# Render the caption/credit area, if one exists. #}
+                    {% if (item.entity.field_az_caption.value) or (item.entity.field_az_credit.value) %}
                       <div class="carousel-caption{% if grid %} az-gallery-caption d-block{% else %} d-none d-md-block{% endif %}">
-                        <p>{{ item.entity.field_az_caption.value }}</p>
+                        {# Render the caption, if one exists. #}
+                        {% if item.entity.field_az_caption.value %}
+                          <p>{{ item.entity.field_az_caption.value }}</p>
```
Comment (danahertzberg) [inline modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig:113]: Can we consider removing the bottom margin here if the caption is alone? Having a caption only adds extra bottom spacing.

**Comment by tadean** [comment]
Applied `az-gallery-caption` for both grid and slider, per @bberndt-uaz suggestion.

File: modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig (hunk 1 above)
Comment (tadean) [reply modules/custom/az_paragraphs/az_paragraphs_photo_gallery/templates/paragraph--az-photo-gallery.html.twig:113]: Agreed, sounds good to me.

**Review by bberndt-uaz — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**